src/runtime/internal/lifecycle.ts

```diff
@@ -31,7 +31,7 @@ export function beforeUpdate(fn: () => any) {
  * 
  * https://svelte.dev/docs#run-time-svelte-onmount
  */
-export function onMount(fn: () => any) {
+export function onMount(fn: (() => () => void) | (() => void | Promise<void>)) {
```

Comment:
```suggestion
export function onMount(fn: (() => () => any) | (() => any | Promise<any>)) {
```

In this case, this will be a type error, right?
But fundamentally, there is no problem.

```svelte
// api.js
export const fetch = () => Promise.resolve({foo: 'bar'});

// Component.svelte
<script>
  import { onMount } from 'svelte';
  import { fetch } from './api';
  onMount(() => fetch())
</script>
```

So IMO, it's better to use `any` type instead of `void`.

Reply:
Thanks for the suggestion but I think that this won't work to prevent a function return from an async onMount.

I am attempting to cause a typescript failure on arguments of type `() => Promise<() => any>`, but the introduction of `() => any` means that this remains a valid argument.

Given I only actually want to disallow this single type, perhaps the following is more appropriate:

```ts
type AnyExcept<Type, Disallowed> = Type & (Type extends Disallowed ? never : Type);
export declare function onMount<T>(fn: () => AnyExcept<T, Promise<() => any>>) {
```

Testing locally, this allows your example above and any other return from onMount **except** for the following:

```js
onMount(async() => { // fails 
   return () => {
      /// anything
   }
})
```